### PR TITLE
Unit tests for backend

### PR DESCRIPTION
**Pull request by JackyChiu**
## Tests 
I added Mocha as our testing framework and Chai as our assertion library. I only added tests for routes that I wrote.

And the workflow around them is that you 
- `describe` the unit your testing
- Tell `it` what it should do
- Tests the results

Going to wait for the merge of previous PR and rebase

## Move article update into new file
Tasks #22

### CODE REVIEW

**Comment by VikramBombhi** [comment]
Can u explain why u split Articles and FetchArticles?

**Comment by JackyChiu** [comment]
@VikramBombhi 
> Can u explain why u split Articles and FetchArticles? Is it simply for the separation of concerns?
Separation of concerns, one is a script that runs every hour, and one is a response handler for `articles` routes

**Comment by JackyChiu** [comment]
I'm going to change the test to expect 10 elements at most instead of always 10 elements

File: test/articles.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                    done()
+                })
+        })
+    })
+})
+
+function checkReponse(res) {
+    res.should.have.status(200)
+    res.body.should.be.arrary
+    res.body.length.should.be.equal(10)
```
Comment (kirinrastogi) [inline test/articles.test.js]: `res.body.length.should.be.at.most(10)`

Comment (JackyChiu) [inline test/articles.test.js]: Changed it 👍

**Comment by kirinrastogi** [comment]
LGTM now

**Pull request merged**